[Failing Test] ci-kubernetes-build-canary does not understand "--platform"

<!-- Please only use this template for submitting reports about continuously failing tests or jobs in Kubernetes CI -->

**Which jobs are failing**:
- `ci-kubernetes-build-canary`

**Which test(s) are failing**:
The overall job

**Since when has it been failing**:

- Sometime between Jan 29 07:00 PST (commit aaadd731f ) and 16:13 PST (commit b184272e2)
- There are some runs that could not be scheduled on a node between these two runs, so it's not clear exactly when it started failing

**Testgrid link**:
https://k8s-testgrid.appspot.com/sig-release-master-informing#build-master-canary

**Reason for failure**:
```
+++ [0201 02:52:52] Building conformance image for arch: amd64
unknown flag: --platform
See 'docker --help'.
```

And then it spits out all the possible Docker options and commands. 

**Anything else we need to know**:
Feels like it may be related to #98529 ? -- That was merged Jan 28 though, and there were successful runs between that and this run

- Spyglass example for failure: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-build-canary/1356060049885106176
- Contrast with Spyglass example for last success: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-build-canary/1355168855353200640

/sig release

I confirm that I hit the same issue in a CI that builds kubernetes using `make quick-release`


```
Makefile:472: recipe for target 'quick-release-images' failed
Failed to build Kubernetes: failed to build images: command "make quick-release-images" failed with error: exit status 2
ERROR: error building node image: failed to build kubernetes: failed to build images: command "make quick-release-images" failed with error: exit status 2
Command Output: +++ [0201 00:13:17] Verifying Prerequisites....
+++ [0201 00:13:17] Building Docker image kube-build:build-c23cf04124-5-v1.15.7-2
+++ [0201 00:13:20] Syncing sources to container
....
+++ [0201 00:17:31] Starting docker build for image: kube-proxy-amd64
unknown flag: --platform
See 'docker --help'.
```

--- a/build/lib/release.sh
+++ b/build/lib/release.sh
@@ -383,7 +383,7 @@ EOF
           echo "COPY nsswitch.conf /etc/" >> "${docker_file_path}"
         fi
 
-        "${DOCKER[@]}" buildx build \
+        DOCKER_CLI_EXPERIMENTAL=enabled "${DOCKER[@]}" buildx build \
           --platform linux/"${arch}" \
           --load ${docker_build_opts:+"${docker_build_opts}"} \
           -q \

--- a/cluster/images/conformance/Makefile
+++ b/cluster/images/conformance/Makefile
@@ -56,7 +56,7 @@ endif
 
 	cd ${TEMP_DIR} && sed -i.back "s|BASEIMAGE|${BASEIMAGE}|g" Dockerfile
 
-	docker buildx build \
+	DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build \
 		--platform linux/${ARCH} \
 		--load \
 		--pull \

